Add(#5): Add oracle db support

indiVar0508 opened the pull request:
Add oracle DB in github action job

indiVar0508 committed:
fix(#5): Skip multi schema tests for oracle DB

Oracle DB requires user in query to drop a particular schema.
currently SQLA seems to generate a query
	`DROP schema schema_name`
but Oracle requires
	`DROP user schema_name`

Tried sqla.schema.DropSchema also still it is not working so skipping
as teardown is not working and remaining tests start failing.

For now, we won't support multi schema testing with oracle (also not
sure if the package works in that scenario)

AbdealiLoKo commented on .github/workflows/main.yml:
```diff
@@ -88,8 +92,55 @@ jobs:
+         - "3.10"
+    services:
+      oracle:
+        image: gvenzl/oracle-xe
```
Which version of oracle does this use ?
Can we use the oldest version available ?

indiVar0508 replied:
It was currently 21, downgraded to 18 for now.

AbdealiLoKo commented on .github/workflows/main.yml:
```diff
@@ -88,8 +92,55 @@ jobs:
+        ports:
+          - 1521:1521
+        env:
+          ORACLE_PASSWORD: p#ssw0rd
```
Can we keep same password for all DBs ?

AbdealiLoKo commented on benchmark.py:
```diff
@@ -53,7 +53,7 @@ class Article(Model):
         __versioned__ = copy(options)
 
-        id = sa.Column(sa.Integer, autoincrement=True, primary_key=True)
+        id = sa.Column(sa.Integer, sa.schema.Sequence("article_id_seq"), autoincrement=True, primary_key=True)
```
Can we use identity instead ?

indiVar0508 replied:
Updated to Identity for all testcases there were certain cases where Identity wasn't working have seperated them out in commits and explained the issue faced.

indiVar0508 replied:
It seems Identity column is causing issues with MSSQL DB where it requires `IDENTITY_INSERT` to be ON

AbdealiLoKo committed:
Add(#5): Add Oracle to github actions

Add Oracle to github actions and test using Pytest
Add connection URL for oracle - if the DB being used is oracle
Add Cx-Oracle requirement in test.txt
Add dependencies for oracle in github actions

indiVar0508 commented:
Closing this PR as Oracle Action job is added as part of #47 and a separate PR will be raised to address failing test case issue for Oracle DB as it has impact on MSSQL DB when tested with Identity object and subquery issue
